File: CHUONG_5/5.15_ViewPager_Exercise/app/src/main/java/com/example/viewpager/ui/fragment/ZodiacItemFragment.java
package com.example.viewpager.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.bumptech.glide.Glide;
import com.example.viewpager.R;
import com.example.viewpager.databinding.FragmentZodiacItemBinding;

public class ZodiacItemFragment extends Fragment {
    private FragmentZodiacItemBinding binding;
    public static final String BUNDLE_KEY_DRAWABLE_ID = "drawableId";

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentZodiacItemBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        loadImage();
    }

    private void loadImage() {
        Bundle args = getArguments();
        if (args != null && args.containsKey(BUNDLE_KEY_DRAWABLE_ID)) {
            int drawableId = args.getInt(BUNDLE_KEY_DRAWABLE_ID);
            binding.image.setImageResource(drawableId);
        } else {
            binding.image.setImageResource(R.drawable.dragon);
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: CHUONG_5/5.15_Zodiac_kotlin/app/src/main/java/com/example/zodiac/ui/fragment/ZodiacItemFragment.kt
package com.example.zodiac.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import com.example.zodiac.R
import com.example.zodiac.databinding.FragmentZodiacBinding
import com.example.zodiac.databinding.FragmentZodiacItemBinding

/**
 * A simple [Fragment] subclass.
 * Use the [ZodiacItemFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class ZodiacItemFragment : Fragment() {
    private var _binding: FragmentZodiacItemBinding? = null
    private val binding get() = _binding!!

    companion object {
        const val BUNDLE_KEY_DRAWABLE_ID = "com.example.drawableId"
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
    
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = FragmentZodiacItemBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        loadImageFromArgs()
    }

    private fun loadImageFromArgs(){
        val drawableId = arguments?.getInt(BUNDLE_KEY_DRAWABLE_ID)
        if (drawableId != null) {
            Glide.with(this).load(drawableId)
                .error(R.drawable.dragon).into(binding.image)
        }
    }
}